### PR TITLE
Aa/benchmark perf improvement

### PR DESCRIPTION
**Pull request by aboodman**

### CODE REVIEW

File: perf/data.ts [hunk 1]
```diff
@@ -50,13 +50,15 @@ export function randomASCIIString(len: number): string {
   return randomStringInternal(len, 128);
 }
 
+// Shared array buffer to reduce allocations.
+const codes: number[] = [];
```
Comment (aboodman) [inline perf/data.ts:54]: In my measurements this function was taking ~30% of execution time on the 16MB benchmark, mostly due to GC. Sharing this buffer reduced the time to ~15%.

File: perf/replicache.ts [hunk 2]
```diff
@@ -359,16 +359,15 @@ export function benchmarkWriteSubRead(opts: {
     async run(bencher: Bencher) {
       const keys = Array.from({length: numKeys}, (_, index) => makeKey(index));
       const sortedKeys = keys.sort();
-      const data: Map<string, TestDataObject> = new Map(
-        keys.map(key => [key, jsonObjectTestData(valueSize)]),
-      );
+      const data: Map<string, TestDataObject> = new Map();
```
Comment (aboodman) [inline perf/replicache.ts:362]: My theory here was that holding onto these random objects for less time would reduce GC pressure, but I'm not 100% sure it did anything.

Comment (grgbkr) [reply perf/replicache.ts:362]: They aren't held for much less time right?  in init below, we put them into replicache, which will keep a reference to them for the full test.

File: perf/replicache.ts [hunk 3]
```diff
@@ -359,16 +359,15 @@ export function benchmarkWriteSubRead(opts: {
...
 
+      const t0 = Date.now();
```
Comment (grgbkr) [inline perf/replicache.ts:364]: remove

File: perf/replicache.ts (hunk 2 above)
Comment (aboodman) [reply perf/replicache.ts:362]: Ah good point, forgot about that. I'll abandon this one.

**Pull request closed without merging**